themes/falconf/src/js/colorChangeOnScroll.js

```diff
@@ -0,0 +1,53 @@
+const SECTIONS = document.querySelectorAll('.scroll-class')
+const body = document.body
+
+const getColorChangeBorder = (offsetTop, sectionHeight) => {
+    return offsetTop + sectionHeight / 2
+}
+
+const isOnTopSection = scrolledFromTop => {
+    return scrolledFromTop < SECTIONS[0].offsetHeight / 2
+}
+
+const isScrolledToBottom = (element, scrolledFromTop) => {
+    const bottom = element.offsetTop + element.offsetHeight
+    return scrolledFromTop === bottom - window.innerHeight 
+}
+
+const getClassName = id => `active-${id}`
+
+const resolveClass = (currentElement, index, scrolledFromTop) => {
+    const isFirst = currentElement.offsetTop === 0 ? true : false
+
+    if (isFirst) {
+        if (isOnTopSection(scrolledFromTop)) {
+            return ''
+        }
+    }
+
+    const isLast = index === SECTIONS.length - 1 ? true : false
+
+    if (isLast) {
+        if (isScrolledToBottom(currentElement, scrolledFromTop))
+            return getClassName(currentElement.id)
+    } else {
+        const top = getColorChangeBorder(currentElement.offsetTop, currentElement.offsetHeight)
+        const nextElement = SECTIONS[index + 1]
+        const bottom = getColorChangeBorder(nextElement.offsetTop, nextElement.offsetHeight)
+        if (scrolledFromTop >= top && scrolledFromTop < bottom) {
+            return getClassName(nextElement.id)
+        }        
+    }
+}
+
+const handleScroll = () => {
+    const scrolledFromTop = window.scrollY
+    SECTIONS.forEach((currentElement, index) => {
+        const newClass = resolveClass(currentElement, index, scrolledFromTop)
+        if(newClass !== undefined) {
+            body.className = newClass
+        }
+    })
+}
+
+window.addEventListener('scroll', handleScroll)
```

Comment:
I would suggest to make this functionality as a module. You will have something like this in the end:

```js
// js/colorChangeOnScroll.js

export default function() {
  // logic that bootstrap and return handleScroll
}

```
And in main file you will trigger it
```js
// index.js

import colorChangeOnScroll from './colorChangeOnScroll.js');

// ...

 $(function (){
  window.addEventListener('scroll', colorChangeOnScroll)
  // ...
 })	 })

```